api/dataset.go

```diff
@@ -382,6 +382,132 @@ func (api *DatasetAPI) addVersion(w http.ResponseWriter, r *http.Request) {
 	log.Debug("upsert version", log.Data{"dataset_id": datasetID, "edition": edition, "version": version})
 }
 
+func (api *DatasetAPI) putDataset(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	datasetID := vars["id"]
+
+	if r.Header.Get(internalToken) != api.internalToken {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
```

Comment:
Consider wrapping the authentication check around the handler instead of having it inside.